### PR TITLE
Feature/seext 8045 ui toolkit use svg assets as icons instead of iconfont

### PR DESCRIPTION
**Pull request by majaklajic**
Introducing breaking changes to existing Icon component:

- Icon component now utilizes SVG icons instead of font icons
- Existing set of font icons is replaced with new SVG icons 
- Users are allowed to add/register custom icons to existing set

### CODE REVIEW

File: components/Icon/Icon.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      fill={color}
+      width={width}
+      height={height}
+      style={{ ...otherStyle }}
+      {...otherProps}
+    />
+  );
+}
+
+Icon.propTypes = {
```
Comment (sstimac) [inline components/Icon/Icon.js:29]: iconProps ?

File: components/Icon/services/icons.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  } else {
+    addNewIcon(config);
+  }
+}
+
+function addNewIcon(config) {
+  const hasName = _.has(config, "name");
+  const hasIcon = _.has(config, "icon");
+
+  if (hasName === false || hasIcon === false) {
```
Comment (sstimac) [inline components/Icon/services/icons.js:30]: !hasName || !hasIcon

**Comment by sstimac** [comment]
@majaklajic let's bump the versions here as well

File: index.js [hunk 3]
```diff
@@ -33,7 +33,7 @@ export { TouchableOpacity } from './components/TouchableOpacity';
 export { TouchableNativeFeedback } from './components/TouchableNativeFeedback';
 export { Touchable } from './components/Touchable';
 export { Button } from './components/Button';
-export { Icon, createIcon } from './components/Icon';
+export { Icon, registerIcons } from "./components/Icon";
```
Comment (toroz) [inline index.js:36]: these should be single quotes, run linter before closing PR

File: theme.js [hunk 4]
```diff
@@ -1181,9 +1183,8 @@ export default (variables = defaultThemeVariables) => ({
 
     backgroundColor: 'transparent',
     color: variables.primaryButtonText.color,
-    textAlign: 'center',
-    textAlignVertical: 'center',
-    fontSize: 24,
+    width: 24,
```
Comment (toroz) [inline theme.js:1186]: do we have a default icon size? (I kinda thought it's 24)

File: components/Icon/assets/index.js [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+  { name: "uber", icon: uber },
+  { name: "unfriend", icon: unfriend },
+  { name: "up-arrow", icon: upArrow },
+  { name: "user-profile", icon: userProfile },
+  { name: "users", icon: users },
+  { name: "video-cam", icon: videoCam },
+  { name: "video-cam-off", icon: videoCamOff },
+  { name: "video-chat", icon: videoChat },
+  { name: "web", icon: web },
+];
```
Comment (toroz) [inline components/Icon/assets/index.js:195]: run linter.

File: components/Icon/services/icons.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  const isObject = _.isObject(config);
+
+  if (!isObject) {
+    return console.warn(
+      "Icon config must be an object or an array of objects."
+    );
+  }
+
+  if (isArray) {
+    return _.forEach(config, function (element) {
```
Comment (toroz) [inline components/Icon/services/icons.js:18]: I don't like the return inconsistency here. 
Currently we have returns for console.warn OR set of icons OR nothing (the 'else' part of this if block).

Also, can we use arrow function here?

File: components/Icon/services/icons.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    addNewIcon(config);
+  }
+}
+
+function addNewIcon(config) {
+  const hasName = _.has(config, "name");
+  const hasIcon = _.has(config, "icon");
+
+  if (hasName === false || hasIcon === false) {
+    return console.warn(`"name" and "icon" keys are required in icon config.`);
```
Comment (toroz) [inline components/Icon/services/icons.js:31]:
```
console.warn(...);
return;
```
We can spare the extra lines of code for consistency in function exits.

File: components/Icon/assets/video-chat.svg [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<!-- Generated by IcoMoon.io -->
+<svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
```
Comment (toroz) [inline components/Icon/assets/video-chat.svg:2]: yes, the defaults are 24 & 24 (answering my previous comment about default sizes)

File: components/Icon/services/icons.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  try {
+    Icons[name] = icon;
+  } catch (error) {
+    console.warn(`Icon "${name}" could not be added: ${error.message}`);
+  }
+}
+
+export function resolveIcon(name) {
```
Comment (toroz) [inline components/Icon/services/icons.js:55]: getIcon?

File: components/Icon/services/icons.js [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  const { name, icon } = config;
+
+  if (_.hasIn(Icons, name)) {
+    return console.warn(`Icon with name "${name}" already exists. Please try using another name.`);
+  }
+
+  if (!_.isString(name)) {
+    return console.warn(`Icon name must be string.`);
```
Comment (toroz) [inline components/Icon/services/icons.js:41]: `Icon name must be a string`

**Pull request merged**